app/javascript/components/student/editor/FileEditor.tsx

```diff
@@ -59,7 +59,7 @@ export const FileEditor = forwardRef<FileEditorHandle, FileEditorProps>(
     )
     const revertContent = useCallback(
       (e) => {
-        setContent(file.content)
+        editorRef.current?.setValue(file.content)
```

Comment:
This was also a crucial line as Monaco editor doesn't behave like a controlled component.